pkg/util/pki/csr_test.go

```diff
@@ -553,3 +559,140 @@ func Test_buildKeyUsagesExtensionsForCertificate(t *testing.T) {
 		})
 	}
 }
+
+func TestSignCSRTemplate(t *testing.T) {
+	// We want to be testing the behavior of SignCSRTemplate in various
+	// contexts; for that, we construct a chain of three certificates: a root
+	// CA, an intermediate CA, and a leaf certificate.
+
+	// We start with a root CA:
+	rootPrivKey, err := GenerateRSAPrivateKey(MinRSAKeySize)
```

Comment:
Great idea to use MinRSAKeySize here, I hadn't even thought to change that :+1:

Reply:
But `MinRSAKeySize` value is 2048, so it is as non-performing as before. 😉 But at least the wish is expressed!